chocolatey/choco
# (#3396) Add specific version check on v3 searches

## Description Of Changes

This updates the handling of searching for packages while at the same time specifying the version that is wanted of the packages.

This ensures that packages will be looked up to verify it has the version the user wants, before returning any results.

## Motivation and Context

When a v3 only feed was being used, no package would be outputted to the console.

## Testing

1. Run the integration tests
2. Copy the contents of the index.json file from one of your sources.
3. Remove the entry in the file mentioning LegacyGallery.
4. Upload the file to a remote location, and update your sources.
5. Run choco search <PACKAGE_NAME> --version <PACKAGE_VERSION> (replace values with known values of a package and older version)
6. Run the same command against your original feed that have a entry for LegacyGallery
7. Compare the results and verify both feeds returned the same information (some metadata may be missing on v3 only feed).

### Operating Systems Testing

- Windows 10

## Change Types Made



## Change Checklist


## Related Issue

Fixes #3396
